src/dstring.c

```diff
@@ -201,13 +204,45 @@ static int dstr_get(d_vm* vm, int a, drax_string* ds) {
   return 1;
 }
 
+static int dstr_touppercase(d_vm* vm, int a, drax_string* ds) {
+  args_fail_required_size(a, 0, "this function does not expect arguments");
+
+  char* upper = (char*) malloc(ds->length);
+
+  int i;
+  for(i = 0; i < ds->length - 1; i++) {
```

Comment:
```suggestion
  for(i = 0; i < ds->length; i++) {
```
we cannot ignore the last character